File: backend/internal/models/full.go
package models

type League struct {
	LeagueID         string              `json:"leagueId"`
	LeagueName       string              `json:"leagueName"`
	Teams            []Team              `json:"teams"`
	Standings        []Standings         `json:"standings"`
	TotalWeeks       int                 `json:"totalWeeks"`
	CurrentWeek      int                 `json:"currentWeek"`
	UpcomingFixtures []Week              `json:"upcomingFixtures"`
	PlayedFixtures   []Week              `json:"playedFixtures"`
	Predict          []PredictedStanding `json:"predict"`
}

File: frontend/src/interfaces/full.ts
import type {PredictedStanding, Standings, Team, Week} from "@/interfaces/league.ts";
import type {MatchResult} from "@/interfaces/simulation";

export interface LeagueData {
    leagueId: string;
    leagueName: string;
    teams: Team[];
    standings: Standings[];
    totalWeeks: number;
    currentWeek: number;
    upcomingFixtures: Week[];
    playedFixtures: Week[];
    predict: PredictedStanding[];
    matches: MatchResult[];
}